tests/MailTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Mail\MailManager;
+use Illuminate\Support\Facades\Event;
+use Stancl\Tenancy\TenancyMailManager;
+use Stancl\Tenancy\Events\TenancyEnded;
+use Stancl\Tenancy\Events\TenancyInitialized;
+use Stancl\Tenancy\Listeners\BootstrapTenancy;
+use Stancl\Tenancy\Listeners\RevertToCentralContext;
+use Stancl\Tenancy\Bootstrappers\MailTenancyBootstrapper;
+
+beforeEach(function() {
+    config(['mail.default' => 'smtp']);
+
+    Event::listen(TenancyInitialized::class, BootstrapTenancy::class);
+    Event::listen(TenancyEnded::class, RevertToCentralContext::class);
+});
+
+test('tenancy swaps the MailManager singleton for an instance of TenancyMailManager', function() {
+    expect(app(MailManager::class))->toBeInstanceOf(TenancyMailManager::class);
+});
+
+test('SMTP mailer transport uses the correct tenant credentials', function() {
+    TenancyMailManager::$tenantMailers = ['smtp'];
+    MailTenancyBootstrapper::$credentialsMap = ['mail.mailers.smtp.password' => 'smtp_password'];
+
+    $tenant = Tenant::create();
+
+    // Initialize tenancy as $tenant and assert that the smtp mailer's transport has the correct password
+    $assertTransportUsesPassword = function(string|null $password) use ($tenant) {
+        tenancy()->initialize($tenant);
+
+        $manager = app(MailManager::class);
+
+        $getMailerViaManager = new ReflectionMethod($manager::class, 'get');
```

Comment:
I think I added `invade()` to the testing dependencies. Could that help you refactor this?

Reply:
Yeah, `invade()` made it much cleaner. I didn't know about that helper

https://github.com/archtechx/tenancy/pull/989/commits/b3cef0c100e4907c1f0d334059c8c2114b90c683

Reply:
Yeah, it's from a Spatie package.

I think we can remove the Reflection-related imports now as well?

Reply:
There are no imports, the Reflection classes didn't need to be imported, so I think we can resolve this

Also, we can refactor [the getDiskPrefix method in BootstrapperTest](https://github.com/archtechx/tenancy/blob/cf5bffea258b59a2c6a80426a371fcceaec30871/tests/BootstrapperTest.php#L362) to use `invade()` too (maybe in another PR)

Reply:
Ah I see, tests are in the global namespace. And yeah, feel free to get rid of all Reflection code in tests 👍🏻